client/hosting-overview/components/plan-card.tsx

```diff
@@ -16,119 +16,164 @@ import useCheckPlanAvailabilityForPurchase from 'calypso/my-sites/plans-features
+					/>
+					<span className="hosting-overview__plan-price-term">
+						{ translate( '/mo', {
+							comment: '/mo is short for per month, referring to the monthly price of a site plan',
+						} ) }
+					</span>
+				</div>
+			) }
+			{ isLoading ? (
+				<LoadingPlaceholder
+					className="hosting-overview__plan-info-loading-placeholder"
+					width="200px"
+					height="16px"
+				/>
+			) : (
+				<div className="hosting-overview__plan-info">{ getBillingDetails() }</div>
+			) }
+			{ isLoading ? (
+				<LoadingPlaceholder
+					className="hosting-overview__plan-info-loading-placeholder"
+					width="200px"
+					height="16px"
+				/>
+			) : (
+				<div
+					className={ classNames( 'hosting-overview__plan-info', {
+						'is-expired': site?.plan?.expired,
+					} ) }
+				>
+					{ getExpireDetails() }
+					<div className="hosting-overview__plan-cta">
+						{ isFreePlan && (
+							<Button primary compact href={ `/plans/${ site?.slug }` }>
+								{ translate( 'Upgrade your plan' ) }
+							</Button>
+						) }
+						{ site?.plan?.expired && (
+							<>
+								<Button compact href={ `/plans/${ site?.slug }` }>
+									{ translate( 'See all plans' ) }
+								</Button>
+								<Button
+									style={ { marginLeft: '8px' } }
+									primary
+									compact
+									href={ `/checkout/${ site?.slug }/${ planData.productSlug }` }
+								>
+									{ translate( 'Renew plan' ) }
+								</Button>
+							</>
+						) }
+					</div>
+				</div>
+			) }
+		</>
+	);
+};
+
+const PlanCard: FC = () => {
+	const translate = useTranslate();
+	const site = useSelector( getSelectedSite );
+	const planDetails = site?.plan;
+	const planName = planDetails?.product_name_short ?? '';
+	const isFreePlan = planDetails?.is_free;
 
 	// Check for storage addons available for purchase.
 	const addOns = AddOns.useAddOns( { selectedSiteId: site?.ID } );
 	const storageAddons = addOns.filter(
 		( addOn ) => addOn?.productSlug === PRODUCT_1GB_SPACE && ! addOn?.exceedsSiteStorageLimits
 	);
 
-	const isLoading = ! pricing || ! planData;
-
 	return (
 		<>
 			<QuerySitePlans siteId={ site?.ID } />
 			<HostingCard className="hosting-overview__plan">
 				<div className="hosting-overview__plan-card-header">
 					<h3 className="hosting-overview__plan-card-title">{ planName }</h3>
 
-					<Button
-						className={ classNames(
-							'hosting-overview__link-button',
-							'hosting-overview__mobile-hidden-link-button'
-						) }
-						plain
-						href={ `/plans/${ site?.slug }` }
-					>
-						{ translate( 'Manage plan' ) }
-					</Button>
+					{ isJetpackProductSlug( planDetails?.product_slug ?? '' ) && (
+						<Button
+							className={ classNames(
+								'hosting-overview__link-button',
+								'hosting-overview__mobile-hidden-link-button'
+							) }
+							plain
+							href={
+								isFreePlan ? `/add-ons/${ site?.slug }` : `/purchases/subscriptions/${ site?.slug }`
```

Comment:
On sites with a paid plan, I think this should link to `/purchases/subscriptions/:site/:planSubscriptionId`. That's what the "Manage plans" buttons does in wordpress.com/plans/:site:

<img width="1274" alt="Screenshot 2024-05-20 at 12 37 18" src="https://github.com/Automattic/wp-calypso/assets/1233880/b640d3f5-ba3d-406c-9106-f6d94a64e07f">

https://github.com/Automattic/wp-calypso/blob/d0246d99e10cb14ae49685639eee719ea247bc2e/client/my-sites/plans/current-plan/purchases-listing.jsx#L227

Similarly, the button should be disabled for users who didn't make the purchase.

Reply:
Thanks for the feedback, I've added the check for `isOwner`.

It will directly hide the Manage button if `! isOwner`.